File: C++/butterfly_pattern.cpp
#include<iostream>
using namespace std;
int main()
{
    int n;
    cout<<"Enter the size of pattern : ";
    cin>>n;
    for(int i=1;i<=n;i++)
    {
        for(int j=1;j<=i;j++){
            cout<<"*";    
        }
        int t=2*n-2*i;
        for(int j=1;j<=t;j++)
        {
            cout<<" ";
        }
        for(int j=1;j<=i;j++)
        {
            cout<<"*";
        }
        cout<<endl;
    }
    for(int i=n;i>=1;i--)
    {
        for(int j=1;j<=i;j++){
            cout<<"*";    
        }
        int t=2*n-2*i;
        for(int j=1;j<=t;j++)
        {
            cout<<" ";
        }
        for(int j=1;j<=i;j++)
        {
            cout<<"*";
        }
        cout<<endl;
    }

}

File: Python/Pattern Printing/butterfly_pattern.py
#!/bin/python3


lenght_of_butterfly = int(input("Enter the number of rows and make sure its even "))

n = lenght_of_butterfly//2

print("Generated butterfly pattern is:\n")

for i in range(1,n+1):
    for j in range(1, 2*n+1):
        if j>i and j< 2*n+1-i:
            print("  ", end="")
        else:
            print(". ", end="")
    print("")

for i in range(n,0,-1):
    for j in range(2*n,0,-1):
        if j>i and j< 2*n+1-i:
            print("  ", end="")
        else:
            print(". ", end="")
    print("")

